plugins/tech-radar/src/api.ts

```diff
@@ -34,52 +45,190 @@ export interface TechRadarApi {
+   *
+   * @remarks
+   *
+   * Supports any value parseable by {@link https://www.npmjs.com/package/color-string | color-string}
+   */
   color: string;
 }
 
+/**
+ * Tech Radar Quadrant which represent area/topic of {@link RadarEntry}
+ */
 export interface RadarQuadrant {
+  /**
+   * ID of the Quadrant
+   */
   id: string;
+  /**
+   * Display name of the Quadrant
+   */
   name: string;
 }
 
+/**
+ * Single Entry in Tech Radar
+ */
 export interface RadarEntry {
-  key: string; // react key
+  /**
+   * React key to use for this Entry
+   */
+  key: string;
+  /**
+   * ID of this Radar Entry
+   */
   id: string;
+  /**
+   * ID of {@link RadarQuadrant} this Entry belongs to
+   */
   quadrant: string;
+  /**
+   * Display name of the Entry
+   */
   title: string;
+  /**
+   * User-clickable URL when rendered in Radar
+   *
+   * @remarks
+   *
+   * You can use `#` if you don't want to provide any other url
+   */
   url: string;
+  /**
+   * History of the Entry moving through {@link RadarRing}
+   */
   timeline: Array<RadarEntrySnapshot>;
+  /**
+   * Description of the Entry
+   */
   description?: string;
 }
 
+/**
+ * State of {@link RadarEntry} at given point in time
+ */
 export interface RadarEntrySnapshot {
+  /**
+   * Point in time when change happened
+   */
   date: Date;
+  /**
+   * ID of {@link RadarRing}
+   */
   ringId: string;
+  /**
+   * Description of change
+   */
   description?: string;
+  /**
+   * Indicates trend compared to previous snapshot
+   */
   moved?: MovedState;
 }
 
+/**
+ * Indicates how {@link RadarEntry} moved though {@link RadarRing} on {@link RadarEntry.timeline}
+ */
+export enum MovedState {
+  /**
+   * Moved down
+   */
+  Down = -1,
+  /**
+   * Didn't move
+   */
+  NoChange = 0,
+  /**
+   * Move up
+   */
+  Up = 1,
+}
+
 /*
  * Types related to data collection for the Radar.
  */
 
+/**
+ * Response from {@link TechRadarApi}
+ */
 export interface TechRadarLoaderResponse {
+  /**
+   * Quadrant of Tech Radar. Should be 4
+   */
   quadrants: RadarQuadrant[];
+  /**
+   * Rings of Tech Radar
+   */
   rings: RadarRing[];
+  /**
+   * Entries visualised in Tech Radar
+   */
   entries: RadarEntry[];
 }
 
 /*
  * Set up the Radar as a Backstage component.
  */
 
+/**
+ * Properties of {@link TechRadarComponent}
+ */
 export interface TechRadarComponentProps {
+  /**
+   * ID of this Tech Radar
+   *
+   * @remarks
+   *
+   * Used when there are multiple Tech Radars and passed to {@link TechRadarApi.load}
+   */
   id?: string;
+  /**
+   * Width of Tech Radar
+   */
   width: number;
+  /**
+   * Height of Tech Radar
+   */
   height: number;
+  /**
+   * Custom React props to the `<svg>` element created for Tech Radar
+   */
   svgProps?: object;
+  /**
+   * Text to filter {@link RadarEntry} inside Tech Radar
+   */
   searchText?: string;
 }
+
+/**
+ * Properties for {@link TechRadarPage}
+ */
+export type TechRadarPageProps = TechRadarComponentProps & {
```

Comment:
Yep, this could use an interface instead without it being breaking, i.e.

```ts
export interface TechRadarPageProps extends TechRadarComponentProps {
  title?: string;
  subtitle?: string;
  pageTitle?: string;
};
```

Whether that's preferred or not is something we haven't really settled on yet. I'm personally beginning to prefer interfaces over types whenever possible though.

Tbh this doesn't quite belong in `api.ts` though, I think we could keep it in the component and export it from there.

Reply:
Interfaces render better IMHO in docs :) I will change that. Will also move `TechRadarComponentProps` and `TechRadarPageProps` to correct files.